Crashes on PyPy3.11 v7.3.20

One example is in doctests for secondzeta(), see e.g. https://github.com/mpmath/mpmath/actions/runs/18441488114/job/52542709682.

```
============================= test session starts ==============================
platform linux -- Python 3.11.13[pypy-7.3.20-final], pytest-8.4.2, pluggy-1.6.0
mpmath backend: gmp
mpmath mp class: <mpmath.ctx_mp.MPContext object at 0x000000002aa94a30>
mpmath version: 1.4.0b2.dev13+g85ec127dd
Python version: 3.11.13 (413c9b7f57f5, Jul 03 2025, 18:03:56)
[PyPy 7.3.20 with GCC 10.2.1 20210130 (Red Hat 10.2.1-11)]
rootdir: /home/runner/work/mpmath/mpmath
configfile: pyproject.toml
testpaths: mpmath, docs
plugins: timeout-2.4.0, hypothesis-6.140.3, cov-7.0.0, xdist-3.8.0
timeout: 600.0s
timeout method: signal
timeout func_only: False
created: 4/4 workers
4 workers [2429 items]

Fatal Python error: Aborted

Stack (most recent call first, approximate line numbers):
  File "/home/runner/work/mpmath/mpmath/mpmath/functions/zeta.py", line 672 in _hurwitz_em
......................................................................[gw2] node down: Not properly terminated
F
replacing crashed worker gw2
........................................................................ [  5%]
[...]

......................................................                   [100%]
=================================== FAILURES ===================================
___________________________ mpmath/functions/zeta.py ___________________________
[gw2] linux -- Python 3.11.13 /opt/hostedtoolcache/PyPy/3.11.13/x64/bin/python
worker 'gw2' crashed while running 'mpmath/functions/zeta.py::mpmath.functions.zeta.secondzeta'
============================= slowest 20 durations =============================
270.54s call     mpmath/tests/test_functions2.py::test_bessel_zeros_extra
102.03s call     mpmath/tests/test_torture.py::test_asymp[<lambda>-150-False23]
81.50s call     mpmath/functions/hypergeometric.py::mpmath.functions.hypergeometric.hyper
65.10s call     mpmath/tests/test_quad.py::test_quadosc
56.08s call     mpmath/tests/test_extra_zeta.py::test_zetazero[241389216-97490234.22767118]
45.91s call     mpmath/tests/test_elliptic.py::test_elliptic_integrals
39.19s call     mpmath/calculus/extrapolation.py::mpmath.calculus.extrapolation.nsum
38.04s call     mpmath/tests/test_format.py::test_mpf_floats_bulk
34.72s call     mpmath/calculus/quadrature.py::mpmath.calculus.quadrature.QuadratureMethods.quadosc
33.06s call     mpmath/tests/test_format.py::test_mpc_complexes
30.56s call     mpmath/ctx_iv.py::mpmath.ctx_iv.MPIntervalContext.siegelz
30.16s call     mpmath/tests/test_basic_ops.py::test_issue_975
29.36s call     mpmath/ctx_fp.py::mpmath.ctx_fp.FPContext.siegelz
28.47s call     mpmath/calculus/quadrature.py::mpmath.calculus.quadrature.QuadratureMethods.quadsubdiv
27.97s call     mpmath/functions/hypergeometric.py::mpmath.functions.hypergeometric.hyp3f2
26.46s call     mpmath/tests/test_torture.py::test_asymp[<lambda>-90-False]
26.39s call     mpmath/functions/zetazeros.py::mpmath.functions.zetazeros.nzeros
25.35s call     mpmath/calculus/quadrature.py::mpmath.calculus.quadrature.QuadratureMethods.quad
24.15s call     mpmath/tests/test_torture.py::test_asymp[barnesg-90-False]
23.53s call     mpmath/tests/test_calculus.py::test_polyroots_legendre_init
=========================== short test summary info ============================
FAILED mpmath/functions/zeta.py::mpmath.functions.zeta.secondzeta
======= 1 failed, 2424 passed, 3 skipped, 2 xfailed in 578.42s (0:09:38) =======
```

Another example is in mpmath/tests/test_torture.py::test_asymp[<lambda>-150-False42], see https://github.com/mpmath/mpmath/actions/runs/18441759191/job/52543336420

```
Fatal Python error: Aborted

Stack (most recent call first, approximate line numbers):
  File "/home/runner/work/mpmath/mpmath/mpmath/libmp/libintmath.py", line 21 in giant_steps
.......s..................[gw1] node down: Not properly terminated
F
replacing crashed worker gw1
```

Probably related to the python-gmp extension (or cpyext, and maybe fixed in nightly builds of the PyPy).

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -32,7 +32,8 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: ['3.10', 3.11, 3.12, 3.13, 3.14, 3.14t, 3.15, 3.15t]
+        python-version: ['3.10', 3.11, 3.12, 3.13, 3.14, 3.14t, 3.15, 3.15t,
+                         pypy3.11]
         nogmpy: [false]
         purepy: [false]
         include:
@@ -40,6 +41,8 @@ jobs:
             nogmpy: true
           - python-version: "3.x"
             purepy: true
+          - python-version: pypy3.11
+            nogmpy: true
           - python-version: pypy3.11
             purepy: true
     env:

